### PR TITLE
A few fixes that make it compile on Mingw64.

### PR DESCRIPTION
**Pull request by lapo-luchini**
Mostly taken from https://github.com/ggerganov/llama.cpp/issues/22

Some might be unnecessary, this is the first version I managed to run.

### CODE REVIEW

**Comment by lapo-luchini** [comment]
Speed doesn't change over WSL Ubuntu version. But the produced executable works for anyone.

File: Makefile [hunk 1]
```diff
@@ -32,7 +32,7 @@ endif
 
 CFLAGS   = -I.              -O3 -DNDEBUG -std=c11   -fPIC
 CXXFLAGS = -I. -I./examples -O3 -DNDEBUG -std=c++11 -fPIC
-LDFLAGS  =
+LDFLAGS  = -static
```
Comment (NouamaneTazi) [inline Makefile:35]: When trying to compile on M1 I get the error
```
I llama.cpp build info: 
I UNAME_S:  Darwin
I UNAME_P:  arm
I UNAME_M:  arm64
I CFLAGS:   -I.              -O3 -DNDEBUG -std=c11   -fPIC -pthread -DGGML_USE_ACCELERATE
I CXXFLAGS: -I. -I./examples -O3 -DNDEBUG -std=c++11 -fPIC -pthread
I LDFLAGS:  -static -framework Accelerate
I CC:       Apple clang version 13.1.6 (clang-1316.0.21.2.5)
I CXX:      Apple clang version 13.1.6 (clang-1316.0.21.2.5)

cc  -I.              -O3 -DNDEBUG -std=c11   -fPIC -pthread -DGGML_USE_ACCELERATE   -c ggml.c -o ggml.o
c++ -I. -I./examples -O3 -DNDEBUG -std=c++11 -fPIC -pthread main.cpp ggml.o utils.o -o main -static -framework Accelerate
ld: library not found for -lcrt0.o
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [main] Error 1
```
But it works when i remove the `-static` flag
And it seems llama.cpp doesn't have this flag here?

**Review by NouamaneTazi — CHANGES_REQUESTED** [review]
LGTM! Thanks for taking care of this 🤗
Left a small comment where the compilation fails for me if you could take a look

File: Makefile (hunk 1 above)
Comment (lapo-luchini) [reply Makefile:35]: I used that to have a standalone `main.exe` not depending on Mingw DLLs, but it only makes sense there I'd say, I'll definitely change it to use an `if`!

**Comment by lapo-luchini** [comment]
Test run on same host, with WSL emulation vs native MingW.

image

WSL Ubuntu:
```
% ./main -s 1 -t 8 -n 64 -p 'Let’s burn some CPU!'
(...)
Let's burn some CPU. Fine prints for six months and the history about 50 years, if we are calling it with documentation, It took a halfway ; you're using that but at work on top of us no use.</s> [end of text]

main: mem per token = 24017564 bytes
main:     load time = 12787.75 ms
main:   sample time =  1001.31 ms
main:  predict time =  9694.05 ms / 220.32 ms per token
main:    total time = 24336.68 ms
```

MingW64:
```
$ ./main -s 1 -t 8 -n 64 -p "Let's burn some CPU."
(...)
Let's burn some CPU. Fine prints for six months and the history about 50 years, if we are calling it with documentation, It took a halfway ; you're using that but at work on top of us no use.</s> [end of text]

main: mem per token = 24017564 bytes
main:     load time =  5325.55 ms
main:   sample time =  1165.75 ms
main:  predict time =  9979.54 ms / 226.81 ms per token
main:    total time = 17367.02 ms
```

**Review by NouamaneTazi — APPROVED** [review]
LGTM! Thanks for the clean PR 🔥

**Pull request merged**